### ISSUE TITLE
Do not force users to have Google's LocationServices enabled

### ISSUE THREAD

**Issue by Schedulaar** [Baseflow/flutter-geolocator#667]
## 🐛 Bug Report

Having turned on location service with Google's location service turned off (see screenshot), I expect to be able to use the location services provided by the system, not by Google's services.
Unfortunately, it seems, that we are stuck with using Google services, if they are installed with the system.

Also it would be nice, to have a hint in the Readme, when and how Google's service is used, and to manually configure whether we should force the user to use google's services.

### Expected behavior
In the scenario above, I expect `await Geolocator.isLocationServicesEnabled()` to return true, but it returns false.

### Reproduction steps

- Turn on location service,
- Turn off Google's location service,
- Run `await Geolocator.isLocationServiceEnabled()`.

### Configuration

**Version:** 6.1.13

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android

![Screenshot_20210215-111147](https://user-images.githubusercontent.com/7025199/107933307-e1376b00-6f7e-11eb-8cca-34a28d5acf26.jpg)

**Comment by mvanbeusekom** [comment]
If you don't want to use the Google services you can use the `forceAndroidLocationManager` boolean in the methods `getLastKnownPosition`, `getCurrentPosition` and `getPositionStream` this will force the geolocator plugin to use the `LocationManager` API instead of the `FusedLocationProviderClient` API (which makes use of Google Services).

Unfortunately this optional parameter isn't available for the `isLocationServicesEnabled` method, which I realise could be an improvement and will be added in version 7.0.0.

**Comment by Schedulaar** [comment]
As far as I can see, the most practical way (from user-perspective) would be, if we use Google's location service and if that's disabled, we use the system location service.
Hence I think (from dev-perspective) it would be best, if Geolocator would have this logic built-in itself (i.e. first check whether Google's service is available; if not check whether System service is available and use the first one available) and only make it optional for the dev to select a preferred method.
This is just a suggestion as this would be the behavior which I would have expected from Geolocator.

**Comment by mvanbeusekom** [comment]
You are right and the geolocator already tries to do so, but as far as I know we can only check if Google Serivces are installed. I am not aware of a possibility to detect if the specific "Google Location Accuracy" setting has been enable or disabled making it hard to automatically switch. 

Of course if can point me to resources which explain how this can be detected I would be happy to update the code.

**Comment by Schedulaar** [comment]
Thanks for your fast response! I just did a quick research and maybe the following resource helps:
https://developers.google.com/android/reference/com/google/android/gms/location/SettingsClient

I am not really into Android development, I just happen to use your library, so maybe you know better than me if this is the right api :)

**Comment by zmGitHub** [comment]
If you use this plug-in in China, you must install Google Service (GMS). You can use TapTap to download the Google suite. If the installation is successful, you must restart your phone, and then remember to put the VPN location and flutter-geolocator on your phone. It’s no problem if you don’t have GMS. The plug-in will always hang without returning

**Comment by irvine5k** [comment]
It's not possible to receive location updates from getPositionStream with Google Location Accuracy turned off, even if `forceLocationManager` is true.

geolocator: ^9.0.2

**Comment by tyypgzl** [comment]
> I can't receive location updates from getPositionStream with Google Location Accuracy turned off, even if `forceLocationManager` is true.
> 
> geolocator: ^9.0.2
> 
> * forceLocationManager **true**, Google Services **on** and Google Location Accuracy **off**:
>   _I don't receive any updates._
> * forceLocationManager **false**, Google Services **on** and Google Location Accuracy **off**:
> 
> ```shell
> E/flutter (31117): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: The location service on the device is disabled.
> E/flutter (31117): #0      GeolocatorAndroid.getPositionStream.<anonymous closure> (package:geolocator_android/src/geolocator_android.dart:187:9)
> ```

same issue, When I set the forceLocationManager value to true, the data comes too late. (sometimes it doesn't come at all).

**Comment by samuCodelers** [comment]
Using the Location plugin has the same problem. When you try to get the position with Location Service ON but Google Location Service (Accuracy) OFF, the position is never received and the apps stuck. I don't know if will be some location plugin that resolve this problem.

**Comment by Tumist76** [comment]
The bug is still present in latest (13.0.1) version. If device has Google Play Services, but disabled Google Location Accuracy, i can't get location even with `forceLocationManager: true`. Is there any updates for this issue?
